### PR TITLE
API changes bring us closer to `google-cloud`

### PR DESCRIPTION
**Pull request by ofrobots**
This change adapts the primary interface of this module to align it
with the API exposed by all the other Google Cloud API Client
Libraries available via the `google-cloud` modules [1].

The primary change is that we return a Debug construtor as the return
value from the module. This, in the future, will provide a Client
Library interface to the Debug APIs. This can be used for programatic
access to list or modify breakpoints, logpoints etc. The constructor,
like other API clients, accepts authentication and other general
configuration.

An instantiated debug object exposes a function `startAgent` that can
be used to activate the process-wide agent that enables the debugging
of an application using the Stackdriver Debug UI in the Google Cloud
console.

Feel free to bike-shed on `startAgent` vs `start`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -70,5 +43,9 @@
     "semver": "^5.1.0",
     "source-map": "^0.5.1",
     "split": "^1.0.0"
+  },
+  "scripts": {
+    "test": "./bin/run-test.sh",
```
Comment (matthewloring) [inline package.json]: If we don't intend to continue using the changelog/publish scripts we should remove them.

File: package.json [hunk 2]
```diff
@@ -52,6 +24,7 @@
     "changelog-maker": "^2.2.2",
     "closure-npc": "*",
```
Comment (matthewloring) [inline package.json]: is it time to drop this?

Comment (ofrobots) [inline package.json]: Yes! but that's not in the scope of this PR.

File: package.json (hunk 1 above)
Comment (ofrobots) [inline package.json]: Yes! but let's now grow the scope of this PR.

File: src/index.js [hunk 3]
```diff
@@ -56,32 +87,32 @@ var initConfig = function(config_) {
...
-    module.exports.private_ = debuglet;
-    start.wasSuccessful_ = true;
+    this.private_ = debuglet;
   }
-}
+};
+
+module.exports = Debug;
 
-setTimeout(function() {
```
Comment (matthewloring) [inline src/index.js]: Do we want to pull out this startup logic as part of this change?

File: package.json (hunk 1 above)
Comment (matthewloring) [inline package.json]: We should preserve all scripts for now then so that we don't have dead code/deps.

**Comment by ofrobots** [comment]
Ready for another look.

File: src/index.js [hunk 4]
```diff
@@ -16,14 +16,45 @@
...
+ * 
+ * This module provides Stackdriver Debugger support for Node.js applications. 
+ * [Stackdriver Debugger](https://cloud.google.com/debug/) is a feature of 
+ * [Google Cloud Platform](https://cloud.google.com/) that lets you debug your
+ * applications in production without stopping or pausing your application.
+ * 
+ * This module provides an agent that lets you automatically enable debugging
+ * without changes to your application. 
+ * 
+ * A low level Client Library for the Debug will also be available through
```
Comment (matthewloring) [inline src/index.js]: Should this be included?

File: src/index.js [hunk 5]
```diff
@@ -16,14 +16,45 @@
...
+ * this module, but is not present at the moment.
+ * 
+ * @constructor
+ * @alias module:debug
```
Comment (matthewloring) [inline src/index.js]: What does the `:` syntax mean?

Comment (ofrobots) [inline src/index.js]: It follows the custom jsdoc parser format used by `google-cloud-node`.

**Pull request merged**